libsplinter/src/biome/key_management/key_store.rs

```diff
@@ -33,8 +34,15 @@ impl PostgresKeyStore {
 }
 
 impl KeyStore<SplinterKey> for PostgresKeyStore {
-    fn add_key(&self, _key: SplinterKey) -> Result<(), KeyStoreError> {
-        unimplemented!()
+    fn add_key(&self, key: SplinterKey) -> Result<(), KeyStoreError> {
+        let key_model = key.into();
+        insert_key(&*self.connection_pool.get()?, key_model).map_err(|err| {
```

Comment:
The insert_key wrapper doesn't seem to be providing much value here (just a level of misdirection). Should this code really be postgresql-specific? (Looks similar to what we will need for sqlite.)

Reply:
the `connection_pool` from the database module is postgres specific